dmriprep/run.py

```diff
@@ -266,6 +266,74 @@ def run_dmriprep_pe(dwi_file, dwi_file_AP, dwi_file_PA, bvec_file, bval_file,
     eddy.inputs.repol = True
     eddy.inputs.niter = 1  # TODO: make this a parameter to the function with default 5
 
+    def id_outliers_fn(outlier_report, threshold, dwi_file):
+        """Get list of scans that exceed threshold for number of outliers
+
+        Parameters
+        ----------
+        outlier_report: string
+            Path to the fsl_eddy outlier report
+
+        threshold: int or float
+            If threshold is an int, it is treated as number of allowed outlier
+            slices. If threshold is a float between 0 and 1 (exclusive), it is
+            treated the fraction of allowed outlier slices before we drop the
+            whole volume. Float param in not yet implemented
+
+        dwi_file: string
+            Path to nii dwi file to determine total number of slices
+
+        Returns
+        -------
+        drop_scans: numpy.ndarray
+            List of scan indices to drop
+        """
+        import nibabel as nib
+        import numpy as np
+        import os.path as op
+        import parse
+
+        with open(op.abspath(outlier_report), 'r') as fp:
+            lines = fp.readlines()
+
+        p = parse.compile(
+            "Slice {slice:d} in scan {scan:d} is an outlier with "
+            "mean {mean_sd:f} standard deviations off, and mean "
+            "squared {mean_sq_sd:f} standard deviations off."
+        )
+
+        outliers = [p.parse(l).named for l in lines]
+        scans = {d['scan'] for d in outliers}
+
+        def num_outliers(scan, outliers):
+            return len([d for d in outliers if d['scan'] == scan])
+
+        if 0 < threshold < 1:
+            threshold *= nib.load(dwi_file).shape[2]
```

Comment:
let's add a TODO to address this part later -- we might have to calculate which dimension corresponds to the z-direction